app/meta.go

```diff
@@ -20,3 +24,21 @@ func SetMetadata(m fyne.AppMetadata) {
 func (a *fyneApp) Metadata() fyne.AppMetadata {
 	return meta
 }
+
+func (a *fyneApp) CustomMetadata(name string) (custom string, exist bool) {
+	custom, exist = customMetadata[name]
+	return
+}
+
+// SetCustomMetadata allow fyne to set the custom metadata map once at the start of your application
+// You shouldn't use it if you do not know what you are doing
+//
+// Since: 2.3
+func SetCustomMetadata(custom map[string]string) bool {
```

Comment:
Do we need a setter here? the other custom data isn't set in this way...
I'm not sure why this blocks a second set, and the need for customMetadataSet.

Reply:
I guess that it comes down to how we want to expose this data. If we are sure that a map is a good solution in all use case and exposing directly the data type is ok, then that would make sense to not have a custom metadata setter. I am not very confident in exposing the map directly and feel like an interface give more freedom.

Reply:
I think there are a few steps between a global setter in the `app` package and directly exposing the `map` in `AppMetadata` are there not?

Reply:
The setter does not enforce the form in which it is stored in the metadata struct. It also doesn't define how it is accessed. Only the way to set it.

Reply:
If it must be set as a `map[string]string` that does rather constrain things, so keeping the flexibility seems to have limited benefit.
Moreover if the flexible approach requires accessing new metadata in a different place in our API I don't see that as being desirable. We should not require a new metadata access point to support custom fields.